lahja/endpoint.py

```diff
@@ -140,43 +120,73 @@ def __init__(self, reader: StreamReader, writer: StreamWriter) -> None:
             raw_size = await self.reader.readexactly(SIZE_MARKER_LENGTH)
             size = int.from_bytes(raw_size, 'little')
             message = await self.reader.readexactly(size)
-            obj = pickle.loads(message)
-            assert isinstance(obj, Message)
+            obj = cast(Message, pickle.loads(message))
             return obj
         except (asyncio.IncompleteReadError, BrokenPipeError, ConnectionResetError):
             raise RemoteDisconnected()
 
 
-class InboundConnection:
+class RemoteEndpoint:
```

Comment:
@cburgdorf thoughts on this change, reverting back to a single connection type.  It seems necessary under the reverse connection paradigm since that allows for two way communication.